qa/tasks/cephfs/test_volumes.py

```diff
@@ -8190,3 +8190,100 @@ def test_volumes_module_finisher_thread(self):
 
         # verify trash dir is clean
         self._wait_for_trash_empty()
+
+class TestToggleVolumes(CephFSTestCase):
+    '''
+    Contains code for enabling/disabling mgr/volumes plugin.
+    '''
+
+    MGR_MOD_NAME = 'volumes'
+    CONFIRM = '--yes-i-really-mean-it'
+
+    def test_enabled_by_default(self):
+        '''
+        Test that volumes plugin is enabled by default and is also reported as
+        "always on".
+        '''
+        output = self.get_ceph_cmd_stdout('mgr module ls')
+        self.assertIn(self.MGR_MOD_NAME, output)
+
+        output = output.split('\n')
+        for line in output:
+            if self.MGR_MOD_NAME in line:
+                self.assertIn('on (always on)', line)
+
+    def test_disable_fails(self):
+        '''
+        Test that running "ceph mgr module disable volumes" fails with EPERM.
+
+        Also test that output of this command suggests user to pass
+        --yes-i-really-mean-it.
+        '''
+        proc = self.run_ceph_cmd(f'mgr module disable {self.MGR_MOD_NAME}',
+                                 stderr=StringIO(), check_status=False)
+        self.assertEqual(proc.returncode, 1)
+
+        proc_stderr = proc.stderr.getvalue()
+        self.assertIn('EPERM', proc_stderr)
+        # ensure that the confirmation flag was recommended
+        self.assertIn(self.CONFIRM, proc_stderr)
+
+    def test_disable_with_confirmation(self):
+        '''
+        Test that running "ceph mgr module disable volumes
+        --yes-i-really-mean-it" successfully disables volumes plugin.
+
+        Also test "ceph mgr module ls" output after this.
+        '''
+        proc = self.run_ceph_cmd(f'mgr module disable {self.MGR_MOD_NAME} '
+                                 f'{self.CONFIRM}', stdout=StringIO())
+        self.assertEqual(proc.returncode, 0)
+
+        output = self.get_ceph_cmd_stdout('mgr module ls')
```

Comment:
Use JSON validation.

Reply:
Since current code is pretty straightforward and does the job efficiently without JSON validation, can we keep it as is?

Reply:
Whenever possible, its better to use JSON parsing.